src/functionalTest/java/uk/gov/hmcts/dts/fact/admin/AdminAuditEndpointTest.java

```diff
@@ -0,0 +1,106 @@
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+import static org.springframework.http.HttpStatus.*;
+import static uk.gov.hmcts.dts.fact.util.TestUtil.*;
+
+
+@ExtendWith(SpringExtension.class)
+public class AdminAuditEndpointTest extends AdminFunctionalTestBase {
+
+    private static final String ADMIN_AUDIT_ENDPOINT = "/admin/audit/";
+
+    private static final String OPENING_TIMES_PATH = "/openingTimes";
+    private static final String ADMINISTRATIVE_COURT_SLUG = "administrative-court";
+    private static final String ADMINISTRATIVE_COURT_OPENING_TIMES_PATH = ADMIN_COURTS_ENDPOINT + ADMINISTRATIVE_COURT_SLUG + OPENING_TIMES_PATH;
+
+    private static final String TEST_HOURS = "test hour";
+    private static final int BAILIFF_OFFICE_OPEN_TYPE_ID = 5;
+    private static final OpeningTime TEST_OPENING_TIME = new OpeningTime(BAILIFF_OFFICE_OPEN_TYPE_ID, TEST_HOURS);
+    private static final String TEST_AUDIT_NAME = "Update court opening times";
+
+    /************************************************************* Get Request Tests. ****************************************************/
+
+    @Test
+    public void shouldReturnAllAudits() throws JsonProcessingException {
+
+        var response = doGetRequest(ADMINISTRATIVE_COURT_OPENING_TIMES_PATH, Map.of(AUTHORIZATION, BEARER + authenticatedToken));
+        final List<OpeningTime> currentOpeningTimes = response.body().jsonPath().getList(".", OpeningTime.class);
+
+        // Action: Adding new opening time
+        List<OpeningTime> expectedOpeningTimes = addNewOpeningTime(currentOpeningTimes);
+        String json = objectMapper().writeValueAsString(expectedOpeningTimes);
+
+        response = doPutRequest(ADMINISTRATIVE_COURT_OPENING_TIMES_PATH, Map.of(AUTHORIZATION, BEARER + authenticatedToken), json);
+        assertThat(response.statusCode()).isEqualTo(OK.value());
+
+        List<OpeningTime> updatedOpeningTimes = response.body().jsonPath().getList(".", OpeningTime.class);
+        assertThat(updatedOpeningTimes).containsExactlyElementsOf(expectedOpeningTimes);
+
+        // Action: Removing the added opening time
+        expectedOpeningTimes = removeOpeningTime(updatedOpeningTimes);
+        json = objectMapper().writeValueAsString(expectedOpeningTimes);
+
+        response = doPutRequest(ADMINISTRATIVE_COURT_OPENING_TIMES_PATH, Map.of(AUTHORIZATION, BEARER + authenticatedToken), json);
+        assertThat(response.statusCode()).isEqualTo(OK.value());
+
+        updatedOpeningTimes = response.body().jsonPath().getList(".", OpeningTime.class);
+        assertThat(updatedOpeningTimes).containsExactlyElementsOf(expectedOpeningTimes);
+
+        final List<Audit> currentAudits = getCurrentAudits();
+        assertThat(currentAudits).isNotEmpty();
+        assertThat(currentAudits.get(currentAudits.size() - 2).getActionDataBefore()).isEqualTo(currentAudits.get(currentAudits.size() - 1).getActionDataAfter());
```

Comment:
We could perhaps only call size() once and also use variables to give the size-1 and size-2 some meaning.